spec/SPEC.md

```diff
@@ -89,259 +79,469 @@ produced the executable.
+skipped.
+
+#### 6.3.5. Input Manifest Construction
+
+A build tool creating an output artifact MUST compute an Input Manifest of
+each mandatory Artifact Identifier Type.
+
+For each input artifact the build tool MUST:
+
+1. Compute the artifact identifier of the input: `${artifact identifier}`
+2. Examine the input for an embedded Artifact ID for an Input Manifest:
+   `${input manifest artifact id}`
+
+The build tool MUST persist an Input Manifest using the
+`${artifact identifier}` and `${input manifest artifact id}` for each input.
+
+#### 6.3.6. Input Manifest Example
+
+```
+gitoid:blob:sha256
+09c825ac02df9150e4f93d12ba1da5d1ff5846c3e62503c814aa3a300c535772
+230f3515d1306690815bd9c3da0d15d8b6fcf43894d17100eb44b6d329a92f61
+2f4a51b16b76bbc87c4c27af8ae062b1b50b280f1ab78e3eec155334588dc88e manifest 4f3a822f776412c049dda53c3277bf2225b51b805ce8a99222af23a7d9f55636
+c71d239df91726fc519c6eb72d318ec65820627232b2f796219e87dcf35d0ab4
+f47ffb3518f236eea6525fd29f057ddd5cda1bb803ccc662e6bc5925afd1e4af
+```
+
+## 7. Storing Input Manifests
+
+This section documents known methods of persisting Input Manifests to
+various stores.
+
+If a build tool persists an Input Manifest to its local filesystem,
+the build tool should write out the Input Manifest to
+`${OMNIBOR_DIR}/manifests/${Artifact Identifier Type URI prefix with ':' replaced by '_'}/${Input Manifest Artifact ID:0:2}/${Input Manifest Artifact ID:2:}`
+where `${Input Manifest Artifact ID}` is the Artifact ID of the Input
+Manifest in lowercase hexadecimal with leading zeros NOT suppressed.
+
+Example:
+
+If `OMNIBOR_DIR=.omnibor` then the Input Manifest with Artifact ID
+`gitoid:blob:sha256:09c825ac02df9150e4f93d12ba1da5d1ff5846c3e62503c814aa3a300c535772`
+would be stored in:
+
+```
+.omnibor/manifests/gitoid_blob_sha256/09/c825ac02df9150e4f93d12ba1da5d1ff5846c3e62503c814aa3a300c535772
+```
+
+### 7.1. Use of a Target Index
+
+Some desirable operations related to Input Manifests require associating an
+Input Manifest with the artifact it is describing. Ideally, the artifact itself
+has embedded in it the Artifact ID of its Input Manifest. However, in cases
+where this is not true, or as a performance optimization even when it _is_ true,
+it is desirable to maintain a "Target Index" which associated the Artifact ID
+of an artifact with the Artifact ID of its Input Manifest.
 
-A build tool creating an output artifact must compute an Input Manifest of
-each mandatory artifact identifier type.
+This Target Index MUST be stored at `${OMNIBOR_DIR}/targets`, and MUST take
+the form of a text file where each line (separated only by the `\n` character)
+MUST be formatted as follows:
 
-For each input artifact the build tool must:
+```
+${Artifact ID of the target artifact}⎵${Artifact ID of the Input Manifest}
+```
+
+`⎵` above refers to the ASCII space character (0x20).
+
+This index file may then be used to improve the performance and reliability of
+operations related to Input Manifests.
+
```

Comment:
Did I miss something here about how to persist these targets in a way we can look them up easily?